simd/unit_tests/include/SIMDTesting_Ops.hpp

```diff
@@ -81,7 +81,9 @@ class absolutes {
   auto on_host(T const& a) const {
     if constexpr (std::is_signed_v<typename T::value_type>) {
 #if defined(KOKKOS_ENABLE_DEPRECATED_CODE_4)
+      KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()
       return Kokkos::Experimental::abs(a);
+      KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_POP()
```

Comment:
Can the compiler decide to ignore the pragma because it's after the return and the code cannot be "executed"?

Reply:
I don't think it would be pruned since it's all preprocessor based and not actually executed